### ISSUE TITLE
mne-qt-browser still opens after running "%matplotlib inline" in IPython / Jupyter session 

### ISSUE THREAD

**Issue by hoechenberger** [mne-tools/mne-qt-browser#73]
Now I know that this package doesn't use Matplotlib, but based on years of "common user experience", what I'd expect is that after I've run the magic `%matplotlib inline`, the `mne-qt-browser` should be disabled (even if set as the default browse backend), and I should get inline Matplotlib-based figures. Same with `%matplotlib widget`, only that the figures are interactive then.

**Comment by larsoner** [comment]
I wouldn't change Qt browser behavior based on the mpl setting, as it's entirely possible someone will want inline mpl figures and Qt based browsing. To me we need to document this somewhere

**Comment by larsoner** [comment]
(specifically that they need to mne.set_browser_backend('matplotlib') as well if they want inline plotting now)

**Comment by hoechenberger** [comment]
But it's a weird and annoying interaction IMHO, because to use `mne-qt-browser`,[ I have to call `%matplotlib qt`](https://github.com/mne-tools/mne-qt-browser#ipython); so there's no easy way to switch between inline Matplotlib and `mne-qt-browser`.

What I mean is that we have a confounding effect between the `%matplotlib` magic and our browser backend setting.

It's just confusing to me.

**Comment by cbrnr** [comment]
Do we really provide tier 1 support for Jupyter notebooks?

**Comment by hoechenberger** [comment]
> Do we really provide tier 1 support for Jupyter notebooks?

IDK, but IPython is what many of our users use (and so do I). It's part of Spyder's and VS Code's interactive Python experience.

**Comment by cbrnr** [comment]
So this affects not just notebooks but IPython in general? Then yes, this is tier 1 (notebooks aren't though IMO).

**Comment by hoechenberger** [comment]
> So this affects not just notebooks but IPython in general?

Seems so. This all uses the same functionality in the "backend", to my knowledge.

**Comment by cbrnr** [comment]
I'm pretty sure the kernel backend is only used for notebooks, and not for the regular IPython (i.e. when you start `ipython` from the terminal).

**Comment by hoechenberger** [comment]
This is what VS Code displays in the interactive window:

<img width="649" alt="Screen Shot 2022-03-29 at 07 38 46" src="https://user-images.githubusercontent.com/2046265/160540909-bb3e0deb-350b-4130-9370-d983ab3ac4bd.png">

To start this interaction mode, `ipykernels` must be installed.

No idea.

In any case, VS Code support is important IMHO 😅

**Comment by cbrnr** [comment]
Yes, that's because you're using a notebook here! I would not recommend doing that, and neither would I support this.

**Comment by hoechenberger** [comment]
No, I'm not using a notebook: this is the VS Code interactive Python window.

**Comment by cbrnr** [comment]
This is effectively using a notebook though. Compare to starting `ipython` from the integrated terminal, this is different (and supposedly everything should just work there).

**Comment by hoechenberger** [comment]
> This is effectively using a notebook though.

Oh, yes, could be that it's technically equivalent.

In any case, it needs to be fixed…

**Comment by cbrnr** [comment]
Yes, it should be fixed, but what I'm saying is that this is not interactive Python anymore. I find notebooks to be a very bad experience, and at least I don't want to put any effort into using and/or fixing things related to that. (In addition, this is probably also using some VS Code specific notebook protocol, which means it doesn't work anywhere else.)

**Comment by hoechenberger** [comment]
> Yes, it should be fixed, but what I'm saying is that this is not interactive Python anymore

I hear what you're saying, but it's effectively irrelevant (please don't take this personally!) as this is just the way VS Code implements the interactive Python experience, and it's what I and many of my colleagues and other MNE users rely on to get their daily work done, so it it's not behaving properly, this is obviously a bug that needs to be fixed.

**Comment by cbrnr** [comment]
I'm not against fixing it. I'm against promoting that as the standard way to get things done.

**Comment by hoechenberger** [comment]
… has anyone even tried how this thing behaves in "native" IPython 😅
(I haven't so far!)

**Comment by cbrnr** [comment]
Well, I'm using normal interactive Python (and/or IPython) all the time, but I haven't tested this specifically. I will do that now.

**Comment by cbrnr** [comment]
OK, I looked at the issue, and since this is specific to notebooks this does not apply to standard interactive Python (there's no `inline` mode). So I guess everything works and it's a notebook issue.

**Comment by hoechenberger** [comment]
Thanks for checking! What about `ipython qtconsole` though?

**Comment by cbrnr** [comment]
That's a notebook running a Jupyter kernel.

**Comment by hoechenberger** [comment]
This is all just technical gibberish to ordinary users …

**Comment by agramfort** [comment]
FYI I had to come up with a "%matplotlib qt" in my vscode lately cause my
qt browser was frozen.
I doubt most users will have this reflex

**Comment by hoechenberger** [comment]
> FYI I had to come up with a "%matplotlib qt" in my vscode lately cause my qt browser was frozen. I doubt most users will have this reflex

This is being tracked in #72
